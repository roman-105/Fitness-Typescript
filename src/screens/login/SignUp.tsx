import React from 'react';
import { ScrollView } from 'react-native';
import { BFButton } from '../../components/Button';
import { Row, Spacer, SelfCenter } from '../../components/Layout/Layout';
import Typography from '../../components/Typography';
import { useNavigation } from '@react-navigation/native';
import Image from 'react-native-fast-image';
import addOnsImage from '../../assets/images/login/add-ons.jpg';
import homeclubsImage from '../../assets/images/login/homeclubs.jpg';
import membershipImage from '../../assets/images/login/membership.jpg';
import styles from './SignUp-style';
import { Routes } from '../../router/routes';
import locale, { country } from '../../utils/i18n/locale';
import BFAnalytics from '../../utils/tools/analytics';
import formatMessage from 'format-message';
import useScrollHeaderShadow from '../../utils/hooks/useScrollHeaderShadow';

const SignUpScreen = () => {
  const navigation = useNavigation();
  const { scroll } = useScrollHeaderShadow({ navigation: navigation });

  function getSignInUrl() {
    return `https://www.basic-fit.com/${locale}-${country ?? 'nl'}/subscription/club`;
  }

  function handleSignUp() {
    BFAnalytics.logEvent('tapSignupButton');
    navigation.navigate(Routes.Webview, { uri: getSignInUrl() });
  }

  return (
    <>
      <ScrollView
        style={styles.flex}
        testID="SignUp"
        contentContainerStyle={styles.container}
        onScroll={(e) => scroll(e)}
        scrollEventThrottle={16}
      >
        <Typography type="h1">{formatMessage('Sign Up')}</Typography>
        <Spacer height={20} />
        <Typography type="regularbfa">
          {formatMessage(
            "Become a Basic-Fit member! In all our clubs (800+) you can find the newest equipment and the free Basic-Fit app gives you endless exercises, training schedules and virtual classes. Choose the membership that suits you the most and add Extra's to make it 100% fit your needs."
          )}
        </Typography>
        <Spacer height={24} />
        <Row marginBottom={8}>
          <Image source={homeclubsImage} style={styles.image} />
          <SelfCenter style={styles.margin}>
            <Typography type="regularbfa" size="m">
              {formatMessage('Select a home club')}
            </Typography>
          </SelfCenter>
        </Row>
        <Row marginBottom={8}>
          <Image source={membershipImage} style={styles.image} />
          <SelfCenter style={styles.margin}>
            <Typography type="regularbfa" size="m">
              {formatMessage('Choose a membership')}
            </Typography>
          </SelfCenter>
        </Row>
        <Row marginBottom={8}>
          <Image source={addOnsImage} style={styles.image} />
          <SelfCenter style={styles.margin}>
            <Typography type="regularbfa" size="m">
              {formatMessage('Add Extraâ€™s')}
            </Typography>
          </SelfCenter>
        </Row>
      </ScrollView>
      <BFButton style={styles.marginButton} onPress={handleSignUp} title="SIGN UP" />
    </>
  );
};

export default SignUpScreen;
