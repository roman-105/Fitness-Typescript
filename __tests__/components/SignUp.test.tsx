import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import * as reactRedux from 'react-redux';
import SignUp from '../../src/screens/login/SignUp';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { useNavigation } from '@react-navigation/native';

let mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      dangerouslyGetState: () => ({ routes: [] }),
      navigate: mockedNavigate,
      setOptions: () => {},
      reset: () => {}
    })
  };
});

const mockStore = configureStore([]);
let store = mockStore({});

describe('SignUp tests', () => {
  beforeEach(() => {
    mockedNavigate = jest.fn();
    jest.useFakeTimers();
    store = mockStore({});
    // @ts-ignore
    jest.spyOn(reactRedux, 'useDispatch').mockImplementation(() => ({
      screenModel: {
        fetch: jest.fn()
      }
    }));
  });

  it('should render just fine', async () => {
    const { toJSON } = render(
      <Provider store={store}>
        <SignUp />
      </Provider>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should open webview on signup tapped', async () => {
    const { getByText } = render(
      <Provider store={store}>
        <SignUp />
      </Provider>
    );
    const button = getByText('SIGN UP');
    fireEvent.press(button);
    const navigation = useNavigation();
    expect(navigation.navigate).toBeCalledWith('WEBVIEW', {
      uri: 'https://www.basic-fit.com/en-nl/subscription/club'
    });
  });
});
