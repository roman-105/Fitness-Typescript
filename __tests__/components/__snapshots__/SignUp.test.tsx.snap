// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp tests should render just fine 1`] = `
Array [
  <RCTScrollView
    contentContainerStyle={
      Object {
        "backgroundColor": "#fff",
        "padding": 20,
      }
    }
    onScroll={[Function]}
    scrollEventThrottle={16}
    style={
      Object {
        "flex": 1,
        "width": "100%",
      }
    }
    testID="SignUp"
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "HeadingProTreble-Heavy",
              "fontSize": 32,
              "fontStyle": "normal",
              "lineHeight": 38,
              "textTransform": "uppercase",
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        Sign Up
      </Text>
      <View
        style={
          Object {
            "height": 20,
            "width": 0,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "fontFamily": "HeadingProDouble-Regular",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "500",
              "letterSpacing": 0.1,
              "lineHeight": 22,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        Become a Basic-Fit member! In all our clubs (800+) you can find the newest equipment and the free Basic-Fit app gives you endless exercises, training schedules and virtual classes. Choose the membership that suits you the most and add Extra's to make it 100% fit your needs.
      </Text>
      <View
        style={
          Object {
            "height": 24,
            "width": 0,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "marginBottom": 8,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "height": 80,
                "width": 80,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={null}
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
              },
              Object {
                "marginLeft": 24,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "HeadingProDouble-Regular",
                  "fontSize": 16,
                  "fontStyle": "normal",
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 22,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Select a home club
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "marginBottom": 8,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "height": 80,
                "width": 80,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={null}
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
              },
              Object {
                "marginLeft": 24,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "HeadingProDouble-Regular",
                  "fontSize": 16,
                  "fontStyle": "normal",
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 22,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Choose a membership
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "marginBottom": 8,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "height": 80,
                "width": 80,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={null}
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
              },
              Object {
                "marginLeft": 24,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "HeadingProDouble-Regular",
                  "fontSize": 16,
                  "fontStyle": "normal",
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 22,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Add Extraâ€™s
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>,
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#7322C9",
        "justifyContent": "center",
        "margin": 20,
        "marginTop": 0,
        "opacity": 1,
        "padding": 8,
      }
    }
    testID="SIGN UP"
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "HeadingProTreble-Regular",
            "fontSize": 32,
            "fontStyle": "normal",
            "lineHeight": 38,
            "textTransform": "uppercase",
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 16,
            },
            undefined,
          ],
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      SIGN UP
    </Text>
  </View>,
]
`;
